public/app/plugins/panel/geomap/utils/checkFeatureMatchesStyleRule.test.ts

```diff
@@ -0,0 +1,127 @@
+import { Feature } from 'ol';
+import { Polygon } from 'ol/geom';
+import { ComparisonOperations } from '../types';
+import { checkFeatureMatchesStyleRule } from './checkFeatureMatchesStyleRule';
+
+describe('check if feature matches style rule', () => {
+  it('can compare with numbers', () => {
+    const rule1 = {
+      operations: ComparisonOperations.EQ,
+      property: 'number',
+      value: 3,
+    };
+    const rule2 = {
+      operations: ComparisonOperations.LT,
+      property: 'number',
+      value: 2,
+    };
+    const rule3 = {
+      operations: ComparisonOperations.LTE,
+      property: 'number',
+      value: 3,
+    };
+    const rule4 = {
+      operations: ComparisonOperations.GT,
+      property: 'number',
+      value: 3,
+    };
+    const rule5 = {
+      operations: ComparisonOperations.GTE,
+      property: 'number',
+      value: 3,
+    };
+
+    const feature = new Feature({
+      geometry: new Polygon([-5e6, 6e6], [-5e6, 8e6], [-3e6, 8e6]),
+      name: 'test polygon',
+    });
+
+    feature.setProperties({ number: 3 });
+
+    expect(checkFeatureMatchesStyleRule(rule1, feature)).toEqual(true);
```

Comment:
rather than have the test definition far away from the results, I like to keep them close togeher -- for example:
```
expect(checkFeatureMatchesStyleRule({
      operations: ComparisonOperations.LT,
      property: 'aaa',
      value: 2,
    }, {
  properties: {
     aaa: 3
  }
})).toEqual(true);

```

Note: we may want to change the signature from "Feature" to "FeatureLike" to help constructing them